syntaxes/happy.YAML-tmLanguage

```diff
@@ -0,0 +1,98 @@
+fileTypes:
+      - y
+name: Haskell Happy
+patterns:
+  - include: "#comments"
+  - include: "#blocks"
+  - include: "#syntax"
+  - include: "#strings"
+repository:
+  block_comment:
+    applyEndPatternLast: 1
+    begin: "{-"
+    captures:
+      '0': { name: punctuation.definition.comment.happy }
+    end: "-}"
+    name: comment.block.happy
+    patterns:
+      - include: "#block_comment"
+  comments:
+    patterns:
+      - name: comment.line.happy
+        match: '\-\-.*$'
+      - include: "#block_comment"
+  blocks:
+    patterns:
+      - name: meta.embedded.block.monad.haskell
+        begin: "{%[%^]?"
+        beginCaptures:
+          "0": { name: punctuation.block.monad.begin.happy }
+        end: "}"
+        endCaptures:
+          "0": { name: punctuation.block.end.happy }
+        patterns:
+          - name: variable.parameter.happy
+            match: \$\d+
+          - name: variable.parameter.field.happy
+            match: \$\$
+          - include: source.haskell
```

Comment:
This doesn't capture nested uses of parameters unfortunately, e.g.:

```haskell
fooBar :: { Something }
       : foo bar { ($1, $2) }
```

This is because the Haskell grammar recursively includes itself to handle content within parentheses (among other situations), whereas we would want it to also include those rules at that point.    

I don't have any idea how we could fix this (short of duplicating the entire Haskell grammar), so it's probably best to just leave this as is (but filing a separate ticket + testcase for this issue).